src/drivers/virtio/virtqueue/packed.rs

```diff
@@ -286,7 +286,7 @@ impl DescriptorRing {
 			}
 
 			if i == 0 {
-				first_ctrl_settings = (ctrl.start, ctrl.buff_id, ctrl.wrap_at_init);
+				first_ctrl_settings = (ctrl.position, ctrl.buff_id, self.drv_wc);
```

Comment:
I think we actually need the start index and the initial wrap counter state here. For multi-descriptor transfers, we delay updating the information for the first descriptor until all the other descriptors are written, so that the device never observes a partial chain. Normally, that is handled by the `WriteCtrl` and  `make_avail`. However, because in batch push we want to delay the update of the flags of the first descriptor not only until the rest of the chain is written but the whole batch is written, we need to keep the `WriteCtrl` state for the first descriptor around.

Reply:
Makes sense. 👍